src/foam/u2/search/TextSearchView.js

```diff
@@ -100,11 +108,27 @@ foam.CLASS({
       name: 'updateValue',
       code: function() {
         var value = this.view.data;
-        this.predicate = ! value ?
-            this.True.create() :
-            this.richSearch ?
-                this.queryParser.parseString(value) :
-                this.ContainsIC(this.property, value);
+
+        if ( ! value ) {
+          this.predicate = this.True.create();
+          return;
+        }
+
+        if ( this.richSearch ) {
+          this.predicate = this.filterController.filters
+            .map((name) => this.of.getAxiomByName(name))
+            .filter((property) => foam.core.String.isInstance(property))
+            .reduce(
```

Comment:
OR is NARY, so you can use .apply() to only create one OR instead of reducing into a tree of OR's.

Reply:
Although, it won't make any difference in the end since FOAM will partialEval it into one OR anyway. Just letting you know.